File: LeetCode/ProblemSet/014.Longest Common Prefix/longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
    let slen = strs.length;
    if (0 === slen) {
        return '';
    }
    if (1 === slen) {
        return strs[0];
    }

    let lcp = '';
    for (let i = 0, len = strs[0].length; i < len; i++) {
        for (let j = 1; j < slen; j++) {
            if (strs[0][i] !== strs[j][i]) {
                return lcp;
            }
        }
        lcp += strs[0][i];
    }
    return lcp;
};

console.log(longestCommonPrefix(["flower", "flow", "flight"]));
console.log(longestCommonPrefix(["dog", "racecar", "car"]));

File: LeetCode/Go-Solutions/ProblemSet/0014.Longest_Common_Prefix/longest-common-prefix.go
package _014_Longest_Common_Prefix

import "sort"

func longestCommonPrefix(strs []string) string {
	sl := len(strs)
	if sl == 0 {
		return ""
	}
	if sl == 1 {
		return strs[0]
	}

	sort.Strings(strs)

	lcp := ""

	for i, l := 0, len(strs[0]); i < l; i++ {
		for si := 1; si < sl; si++ {
			if strs[0][i] != strs[si][i] {
				return lcp
			}
		}
		lcp += string(strs[0][i])
	}
	return lcp
}
